src/stan-users-guide/latent-discrete.Rmd

```diff
@@ -1019,3 +1019,182 @@ the probability mass function of $z_i$ in the posterior.   These may
+interested in functions of both the discrete and continuous parameters. 
+Luckily, in practice we are commonly interested in expectations involving only 
+one of either the continuous or discrete parameters, for example the posterior
+mean of an individual scalar component of these parameters. In these cases,
+evaluating $\textsf{E}(g(X, Y) \mid X)$ becomes tractable.
+
+If $g(X, Y)$ is a function of only the continuous parameters $X$, the 
+conditional expectation simplifies:
+
+$$
+\textsf{E}(g(X, Y) \mid X) 
+= \textsf{E}(g(X) \mid X) = g(X).
+$$
+
+Our conditional estimator therefore becomes:
+
+$$
+\frac{1}{M} \sum_{i = 1}^M g(x_i).
+$$
+
+This makes performing inference straightforward. First, we write a Stan program 
+that implements the marginalized posterior $p_X(x)$. Running this model will 
+produce samples from the marginalized posterior. We then evaluate $g(\cdot)$ at
+each sample, and use the average over the samples to estimate the expectation 
+of interest. That is, we use Stan `as usual', except that we implement a
+marginalized posterior.
+
+### Recovering the discrete parameters {-}
+
+For the discrete parameters, we are often only interested in the marginal 
+posterior mass function.  For example, for a univariate $Y$ we want to 
+calculate $\Pr(Y = k)$ for all possible values of $k$; note that this allows 
+us to calculate the posterior of any quantity of interest involving $Y$. To 
+estimate $\Pr(Y = k)$ under the marginalization scheme we can note that
+
+$$
+\Pr(Y = k) = \textsf{E}(\mathbf{1}(Y = k)),
+$$
+
+in other words that $g(X, Y) = \mathbf{1}(Y = k)$. Then, from the properties of conditional expectation we have
+
+$$
+h(X) 
+= \textsf{E}(g(X, Y) \mid X) 
+= \textsf{E}(\mathbf{1}(Y = k) \mid X) 
+= \Pr(Y = k \mid X).
+$$
+
+We can therefore estimate $\Pr(Y = k)$ with
+
+$$
+\frac{1}{M} \sum_{i = 1}^M \Pr(Y = k \mid X = x_i).
+$$
+
+When calculating this conditional probability it is important to remember that
+we are also conditioning on the observed data, $D$. That is, we are really
+estimating $\Pr(Y = k \mid D)$ with
+
+$$
+\frac{1}{M} \sum_{i = 1}^M \Pr(Y = k \mid X = x_i, D).
+$$
+
+This point is important as it suggests one of the main ways of calculating the required conditional probability. Using Bayes' theorem gives us:
+
+$$
+\Pr(Y = k \mid X = x_i, D) 
+= \frac{               \Pr(D \mid Y = k, X = x_i) \Pr(Y = k \mid X = x_i)}
+       {\sum_{k = 1}^K \Pr(D \mid Y = k, X = x_i) \Pr(Y = k \mid X = x_i)}.
+$$
+
+Here, $\Pr(D \mid X = x_i, Y = k)$ is the likelihood conditional on a 
+particular value of the latent variables. Crucially, all elements of the 
+expression can be calculated using the samples from the posterior of the 
+continuous parameters and knowledge of the model structure. Note that for 
+certain models, such as [zero-inflated count models](#zero-inflated.section), 
+some values of the latent parameters and data may be incompatible, giving probabilities of exactly zero.
```

Comment:
I don't understand where the zero probabilities come from with zero inflation?  Do you mean numerically zero because of underflow?

Reply:
If the data are positive and then the latent discrete zero-inflation state (i.e. the parameter that has been marginalized out) cannot be one (where one corresponds to the zero-inflated state).

Reply:
Thanks, @jsocolar.  I tend not to think of those boundary parameter estimates as they only arise in Stan due to underflow or rounding of inverse logit.  That is, we assume parameters on a (-inf, inf) scale and then transform using inverse logit to (0, 1).  But given the vagaries of floating point, we can underflow to 0 or round to 1 when the values are too extreme.

Reply:
@bob-carpenter This isn't a boundary issue.  The parameter is a binary discrete parameter that gets marginalized out of the model.  It's the discrete parameter that we're trying to recover.  Where the data are nonzero, the posterior is a point mass at one.  When the data are zero, then there's some posterior mass at both one and zero.